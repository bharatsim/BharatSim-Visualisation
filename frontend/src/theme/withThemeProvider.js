import React from 'react';
import { MuiThemeProvider, CssBaseline } from '@material-ui/core';

import theme from './theme';

function withThemeProvider(WrappedComponent) {
  return function WithTheme(props) {
    return (
      <MuiThemeProvider theme={theme}>
        <CssBaseline />
        <WrappedComponent {...props} />
      </MuiThemeProvider>
    );
  };
}

export default withThemeProvider;
