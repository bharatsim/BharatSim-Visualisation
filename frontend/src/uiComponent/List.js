import React from 'react';
import PropTypes from 'prop-types';
import { Box, List as MuiList, ListItem } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  mainContainer: {
    border: `1px solid ${theme.colors.tableBorder}`,
    borderRadius: theme.spacing(1),
    background: theme.palette.background.paper,
    height: theme.spacing(32),
    paddingBottom: theme.spacing(1),
    maxWidth: theme.spacing(60),
    width: '100%',
  },
}));

const useListItemStyles = makeStyles((theme) => ({
  root: {
    ...theme.typography.caption,
    lineHeight: 1,
    padding: theme.spacing(1, 2),
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: `${theme.colors.primaryColorScale['50']} !important`,
    },
  },
}));

const useListStyles = makeStyles((theme) => ({
  root: {
    padding: theme.spacing(1, 0),
    height: '100%',
    overflow: 'auto',
  },
}));

// const Input = withStyles((theme) => ({
//   root: {
//     position: 'relative',
//     height: theme.spacing(7),
//     borderBottom: `1px solid ${theme.colors.grayScale['200']}`,
//     borderRadius: 0,
//     '&:hover $notchedOutline': {
//       border: 0,
//     },
//     '@media (hover: none)': {
//       '&:hover $notchedOutline': {
//         border: 0,
//       },
//     },
//     '&$focused $notchedOutline': {
//       border: 0,
//     },
//     ...theme.typography.body1,
//   },
//   notchedOutline: {
//     border: 0,
//   },
//   input: {
//     paddingTop: theme.spacing(1),
//     paddingBottom: theme.spacing(1),
//   },
//   adornedStart: {
//     paddingLeft: theme.spacing(1),
//   },
// }))(OutlinedInput);

function List({ listItem, onClick, listItemProps }) {
  const classes = useStyles();
  const listItemClasses = useListItemStyles();
  const listClasses = useListStyles();
  return (
    <Box className={classes.mainContainer}>
      <MuiList classes={listClasses}>
        {listItem.map((item, index) => {
          const key = `index-${index}`;
          return (
            <ListItem
              {...listItemProps}
              classes={listItemClasses}
              onClick={() => onClick(item)}
              key={key}
            >
              {item}
            </ListItem>
          );
        })}
      </MuiList>
    </Box>
  );
}

List.defaultProps = {
  listItemProps: {},
};

List.propTypes = {
  listItem: PropTypes.arrayOf(PropTypes.string).isRequired,
  onClick: PropTypes.func.isRequired,
  listItemProps: PropTypes.shape({}),
};

export default List;
