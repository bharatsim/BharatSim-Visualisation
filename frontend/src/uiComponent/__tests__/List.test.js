import React from 'react';
import { render } from '@testing-library/react';
import { fireEvent } from '@testing-library/dom';
import List from '../List';
import withThemeProvider from '../../theme/withThemeProvider';

const ListWithProvider = withThemeProvider(List);

describe('<List />', () => {
  it('should match snapshot with list item', () => {
    const { container } = render(
      <ListWithProvider onClick={jest.fn()} listItem={['field1', 'field2']} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('should call onClick callback on click of any list item', () => {
    const onClick = jest.fn();
    const { getByText } = render(
      <ListWithProvider onClick={onClick} listItem={['field1', 'field2']} />,
    );

    fireEvent.click(getByText('field1'));

    expect(onClick).toHaveBeenCalled();
  });
});
