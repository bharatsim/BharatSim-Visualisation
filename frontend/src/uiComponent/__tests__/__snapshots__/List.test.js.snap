// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should match snapshot with list item 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-4 makeStyles-mainContainer-1"
  >
    <ul
      class="MuiList-root makeStyles-root-3 MuiList-padding"
    >
      <li
        class="MuiListItem-root makeStyles-root-2 MuiListItem-gutters"
      >
        field1
      </li>
      <li
        class="MuiListItem-root makeStyles-root-2 MuiListItem-gutters"
      >
        field2
      </li>
    </ul>
  </div>
</div>
`;
